### ISSUE TITLE
Add @taskgroup decorator

### ISSUE THREAD

**Issue by casassg** [apache/airflow#11870]
**Description**

Similarly to `@dag` and `@task` decorators, we can add a decorator to easily generate `TaskGroup` instances. 

**Use case / motivation**

This would be used for reusable pieces of DAGs. An example may be to hyperparameter tune an ML pipeline. You may want to generate several parallel subpipelines with different settings/parameters to be executed in parallel. At the end pull the result and decide the best parametrization.  

```py
@task
collect_dataset(...)
  pass

@task
def train_model(...)
   pass

@task_group
def movielens_model_pipeline(learning_rate: int, feature_list: list, dataset: XComArg):  
  dataset = filter_dataset(dataset, feature_list)
  train_model(dataset, learning_rate)

@dag
def movielens_hpt(dataset_path: str, feature_list:list=['year', 'director']):
  dataset = load_dataset(dataset_path)
  for i in range(0.1, 0.9, step=0.1):
    movielens_model_pipeline(i, feature_list, dataset)
  
  decide_best_model(...)
```

**Related Issues**

https://github.com/apache/airflow/issues?q=is%3Aopen+is%3Aissue+label%3AAIP-31

**Comment by ryw** [comment]
Sounds good to me 👍

**Comment by thcidale0808** [comment]
@ryw can I implement this feature?

**Comment by ryw** [comment]
@thcidale0808 sorry for slow reply - looks like someone else is working on it here - https://github.com/apache/airflow/pull/12159

**Comment by thcidale0808** [comment]
thanks @ryw. I'll look for another one to contribute. @kaxil maybe you should assign this issue to VBhojawala.

**Comment by VBhojawala** [comment]
Closing current pull request to re-raise pull request.
